### ISSUE TITLE
Settings not saving

### ISSUE THREAD

**Issue by bbuchan10** [qooob/authentic-theme#272]
I am having an issue where my settings are not saving in the theme settings. Such as setting the option for "Check for Authentic Theme updates". It has been happening for many versions now. Same for theme and logos. Theme and Logos stay until i log out and back in.

**Comment by qooob** [comment]
Make sure that `/etc/webmin/authentic-theme` is a directory and have proper (writable) permissions.

**Comment by bbuchan10** [comment]
/etc/webmin/authentic-theme is root:root 0777 and there is a file "config" in with with root:root 0644

**Comment by qooob** [comment]
That sounds like system specific. Check, when you logout, do files in that directory disappear? Files like settings.js, logo.PNG and other? Are the files there in current session?

**Comment by bbuchan10** [comment]
i run Ubuntu 14.04.01 and the only thing in /etc/webmin/authentic-theme is the config file

**Comment by qooob** [comment]
And after you save settings? Go to console and see what you have in the directory after saving?

**Comment by bbuchan10** [comment]
when i save settings. nothing changes in the directory. still only config file.

**Comment by qooob** [comment]
Then you should provide me with log information. I'm at the moment have not clue why it's happening on your side.

**Comment by bbuchan10** [comment]
Ok, what logs do you need and how do i get to them?

**Comment by qooob** [comment]
You should debug Perl and take a look at Webmin's `miniserv.log`.

**Comment by qooob** [comment]
What is the output of `ls -lsaZ /etc/webmin/authentic-theme`

**Comment by bbuchan10** [comment]
ls -lsaZ /etc/webmin/authentic-theme
total 12
4 drwxrwxrwx   2 root root ? 4096 Oct 17 06:37 .
4 drwxr-xr-x 144 root root ? 4096 Oct 16 10:02 ..
4 -rw-r--r--   1 root root ?  114 Oct 16 10:02 config

**Comment by qooob** [comment]
..and what browser/OS are you using? (client-side)

**Comment by bbuchan10** [comment]
chrome, windows 10

On Wed, Oct 21, 2015 at 2:16 PM, Ilia Rostovtsev <notifications@github.com>
wrote:

> ..and what browser/OS are you using? (client-side)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/qooob/authentic-theme/issues/272#issuecomment-149998044>
> .
>



-- 


Bryan Buchan
Youtube Channel: Monkeybdb <http://www.youtube.com/user/monkeybdb>

**Comment by bbuchan10** [comment]
i also confirmed that it happens on IE, Edge and Firefox

**Comment by qooob** [comment]
@bbuchan10 Did you try settings permissions to `drwxr-xr-x` (755) for `/etc/webmin/authentic-theme`?

**Comment by specktator** [comment]
@bbuchan10 @qooob Guys I'm using authentic theme on Ubuntu 14.04.3 LTS and it's fine. No problem with settings etc. Try to re-install the theme.

**Comment by bbuchan10** [comment]
I have deleted and reinstalled the theme multiple times. It has worked in
the past and it works fine on my other servers.

**Comment by qooob** [comment]
It is not Authentic Theme issue. I can not know why settings are not saved on that one particular machine. 

Please report back to me, when you find out.

**Comment by bbuchan10** [comment]
would .old'ing the folder then reinstalling the theme, restore the folder with proper permissions?

**Comment by qooob** [comment]
You could try. It could help.

**Comment by bbuchan10** [comment]
Sorry to revive this old thing, but i am still having the issue and .old the folder did not help. however i may have some useful info. When i look at /etc/webmin/authentic-theme/ i only see the config file. however on a working server i see config, settings, and my logo files. when i transplant the files, everything worked great until i try to change the settings again and the files are deleted, and the folder is just the config file again. The modified time has changed as well on the folder so it is not a permissions issue. (i also confirmed those with a working server).

**Comment by qooob** [comment]
Which server side OS do you have issue in? What client side OS/browser do you use?

**Comment by bbuchan10** [comment]
happens on any broswer and i use Ubuntu 14.04.3

**Comment by qooob** [comment]
Open Chrome console, click Network tab, filter XHR go to settings and click Save. You shoukd see the request's respond! What is that?

**Comment by bbuchan10** [comment]
/index.cgi?xhr-settings=1&save=1 response just shows "this response has no request data available"

**Comment by bbuchan10** [comment]
the working server shows the same thing, so i may not have gotten the info you wanted.

**Comment by qooob** [comment]
How are is it? Are the settings saved now?

**Comment by bbuchan10** [comment]
Nope, I'm on .31 too

**Comment by bbuchan10** [comment]
Btw, I'm on virtualmin, don't know if that matters

**Comment by qooob** [comment]
What do you meant, you're on Virtualmin?

**Comment by bbuchan10** [comment]
I meant I'm not using just Webmin, I also have Virtualmin installed.

**Comment by qooob** [comment]
It's normal. I tested it in many different situations.

**Comment by qooob** [comment]
Alright, I have found very strange Perl behavior on comparing two files. With new 17.50 your issue most likely will be resolved. Please, let me know after 17.50 is released, if your settings are saved!?

Thanks!

**Comment by bbuchan10** [comment]
unfortunately, it appears that when i save settings the files are still
disapearing in the authentic-theme folder.

On Thu, Jan 21, 2016 at 2:51 AM, Ilia Rostovtsev <notifications@github.com>
wrote:

> Alright, I have found very strange Perl behavior on comparing two files.
> With new 17.50 your issue most likely will be resolved. Please, let me know
> after 17.50 is released, if your settings are saved!?
>
> Thanks!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/qooob/authentic-theme/issues/272#issuecomment-173502814>
> .
>



-- 


Bryan Buchan
Youtube Channel: Monkeybdb <http://www.youtube.com/user/monkeybdb>

**Comment by qooob** [comment]
They should be saved to `/etc/webmin/authentic-theme` and then moved to `/usr/webmin/authentic-theme/unauthenticated/js`. (RHEL) On Ubuntu it's `share` folder.. Not sure how it's possible otherwise!?

**Comment by thehutman** [comment]
We had this same problem, and today I traced it down to having usermin installed, and having a link in the /etc/usermin folder that pointed to the authentic-them config in the /etc/webmin folder.  This used to be a recommended tactic to handle sharing settings in usermin from webmin.

I removed the link under usermin, created the authentic-theme folder in usermin, copied over the "config" file from webmin to usermin, and then files started to save normally.
